src/main/java/com/celements/web/plugin/cmd/ExternalJavaScriptFilesCommand.java

```diff
@@ -188,11 +191,19 @@ public String getAllExternalJavaScriptFiles() throws XWikiException {
           context.getDatabase(),
           context.getDoc().getDocumentReference().getLastSpaceReference().getName(),
           "WebPreferences"), context));
-      addAllExtJSfilesFromDoc(context.getDoc());
-      XWikiDocument pagetype = getPageTypeDoc(context.getDoc());
-      if (pagetype != null) {
-        addAllExtJSfilesFromDoc(pagetype);
+      PageTypeReference pageTypeRef = getPageTypeResolver().getPageTypeRefForDocWithDefault(
+          context.getDoc().getDocumentReference());
+      DocumentReference pageTypeDocRef = new DocumentReference(context.getDatabase(), "PageTypes",
```

Comment:
please use XObjectPageTypeUtilsRole component to resolve the PageTypeReference to a DocumentReference